syntax = "proto3";

service ShardEnvoy {
  rpc Start (UsrIns) returns (UsrOuts) {}
  rpc Receive (ShardOuts) returns (Empty) {}
}

message UsrIns {
  string prompt = 1;
  int32 max_tokens = 2;
}

message UsrOuts {
  double prompt_time = 1;
  int32 prompt_t_cnt = 2;
  double gen_time = 3;
  int32 gen_t_cnt = 4;
  string response = 5;
}

message ShardOuts {
  string url = 1;
  int32 layer_num = 2;
  bytes data = 3;
  bytes arr_map = 4;
}

message Empty {}
